### DISCUSSION TITLE
How to use env variables in the middleware?

### DISCUSSION THREAD

**Discussion by glung** [expatfile/next-runtime-env#78] [Q&A] [answered]
The package claims to [work with the middleware](https://github.com/expatfile/next-runtime-env/blob/abe918541bb7fcbe6c259c6d0dd04ca203ff8952/README.md?plain=1#L7). 

Though, in my experience (see below), I could not get it to work. Considering the middleware is an Edge runtime, that probably makes sense. Is there some other mechanism to support this use case?

---

```
SERVER_ENV=server NEXT_PUBLIC_ENV=client_server next dev
```

```
import { env } from "next-runtime-env";
import { NextRequest, NextResponse } from "next/server";

export async function middleware(req: NextRequest) {
  console.log("SERVER_ENV", env("SERVER_ENV));
  console.log("NEXT_PUBLIC_ENV", env("NEXT_PUBLIC_ENV"));

[...]
}
````

**Comment by HofmannZ** [comment, marked as answer]
THis was a bug in Next.js itself. Please update Next.js to the latest version. 🙂

**Reply by jpsaturnino** [reply]
I'm using `next: 12.3.4` the last one available for 12.x, but in the middleware my env show as undefined
![image](https://github.com/expatfile/next-runtime-env/assets/47997386/64cc4213-7f28-4fd5-a23c-97ef222a77a1)

any suggestion?

**Reply by jpsaturnino** [reply]
@glung It works using process.env.VAR instead of env('VAR') on middleware file. Here the readme of 1.x branch
![image](https://github.com/expatfile/next-runtime-env/assets/47997386/ebf0d1ea-e80c-4dcb-ab7c-003a52675d2d)

**Reply by wujekbogdan** [reply]
@HofmannZ 

> THis was a bug in Next.js itself. Please update Next.js to the latest version.

What's the Next.js version that fixes the bug? Could you point me to the changelog?

I'm still experiencing this bug with Next.js `13.3.4` and `next-runtime-env@1.7.4`. The reason I use the `1.x` version of the library is that the app I'm working on still uses the Page Router.

```
console.log(env('NEXT_PUBLIC_MY_VAR')) // logs undefined
console.log(process.env.NEXT_PUBLIC_MY_VAR) // logs the proper value
```
